### PR TITLE
Update documentation and upgrade software to conform to this documentation

### PR DESCRIPTION
**Pull request by mseaton**

### CODE REVIEW

File: src/main/java/org/pih/petl/job/schedule/PetlScheduledExecutionTask.java [hunk 1]
```diff
@@ -49,20 +50,21 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
...
-                    else {
-                        Date lastStartDate = latestExecution.getStarted();
-                        log.debug("Last Execution Start: " + lastStartDate);
-                        Date lastEndDate = latestExecution.getCompleted();
-                        log.debug("Last Execution End: " + lastEndDate);
-                        if (lastEndDate != null) {
-                            PetlJobConfig jobConfig = jobs.get(jobPath);
-                            Schedule schedule = jobConfig.getSchedule();
-                            if (schedule != null) {
+                    PetlJobConfig jobConfig = jobs.get(jobPath);
```
Comment (mseaton) [inline src/main/java/org/pih/petl/job/schedule/PetlScheduledExecutionTask.java:53]: The purpose of this change is to only run a job if that job has a schedule defined.  This is necessary in order for the "job-pipeline" job type to work correctly, as we typically only want the jobs that it refers to to be executed within the job-pipeline, not as independent jobs.

File: pom.xml [hunk 2]
```diff
@@ -68,6 +68,8 @@
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.22</version>
```
Comment (mseaton) [inline pom.xml:71]: I was running into errors related to mysql version, and this fixed it.

File: pom.xml [hunk 3]
```diff
@@ -113,7 +115,13 @@
...
             <groupId>pentaho-kettle</groupId>
             <artifactId>kettle-engine</artifactId>
-            <version>7.1-SNAPSHOT</version>
+            <version>9.1.0.0-324</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.pentaho.di.plugins</groupId>
+            <artifactId>pdi-core-plugins-impl</artifactId>
+            <version>9.1.0.0-324</version>
```
Comment (mseaton) [inline pom.xml:124]: After upgrading mysql connector, I ran into issues with Pentaho 7, which were fixed in Pentaho 9, so I upgraded.  I also found that in Pentaho 9 some of the transforms that were in use in Malawi needed to be pulled in from a separate plugin library, so I tracked that down and added it here.

File: src/main/java/org/pih/petl/job/type/PentahoJob.java [hunk 4]
```diff
@@ -68,10 +67,10 @@ public void execute(final ExecutionContext context) throws Exception {
         System.setProperty("KETTLE_HOME", jobDir.getAbsolutePath());
 
         try {
-            // Configure kettle.properties with PIH_PENTAHO_HOME which we aquire from the configuration
-            File srcDir = appConfig.getPetlHomeDir();
+            // Configure kettle.properties with PIH_PENTAHO_HOME which we acquire from the configuration
+            File srcDir = new File(configuration.getProperty(PIH_PENTAHO_HOME, jobFile.getParent()));
```
Comment (mseaton) [inline src/main/java/org/pih/petl/job/type/PentahoJob.java:71]: This was necessary to allow the pih-pentaho codebase to be loaded from someplace outside of the petl install directory, configurable by server as needed.

File: src/main/java/org/pih/petl/job/type/PentahoJob.java [hunk 5]
```diff
@@ -87,6 +86,7 @@ public void execute(final ExecutionContext context) throws Exception {
             try {
                 log.debug("Initializing Kettle Environment");
                 log.debug("KETTLE_HOME = " + System.getProperty("KETTLE_HOME"));
+                System.setProperty("KETTLE_PLUGIN_CLASSES", "org.pentaho.di.trans.steps.append.AppendMeta");
```
Comment (mseaton) [inline src/main/java/org/pih/petl/job/type/PentahoJob.java:89]: This is the class that was moved to a plugin in pentaho 9, and this is what you need to do to tell Kettle to load it.

File: src/main/java/org/pih/petl/job/type/SqlServerImportJob.java [hunk 6]
```diff
@@ -137,14 +138,19 @@ public void execute(final ExecutionContext context) throws Exception {
...
+                        }
+                        else {
                             log.debug("This is the last statement, treat it as the extraction query");
+                            statement = sourceConnection.prepareStatement(
+                                    sqlStatement, ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY
+                            );
                             ResultSet resultSet = null;
                             try {
-                                resultSet = statement.getResultSet();
+                                resultSet = ((PreparedStatement)statement).executeQuery();
```
Comment (mseaton) [inline src/main/java/org/pih/petl/job/type/SqlServerImportJob.java:153]: The "resultSet.last()" mechanism used below no longer works by default with the MySQL 8 connector.  You need to instead use a prepared statement and instruct it to allow resultset scrolling.  This is the nature of the above changes.

File: src/main/resources/kettle-password-encoder-plugins.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<password-encoder-plugins>
+
+    <password-encoder-plugin id="Kettle">
+        <description>Kettle Password Encoder</description>
+        <classname>org.pentaho.support.encryption.KettleTwoWayPasswordEncoder</classname>
+    </password-encoder-plugin>
+
+</password-encoder-plugins>
```
Comment (mseaton) [inline src/main/resources/kettle-password-encoder-plugins.xml:8]: This file is newly required as of Pentaho 9.

**Review by mogoodrich — APPROVED** [review]
LGTM!  

I didn't review the entire README, but looked at the code changes and will keep an eye out for any problems after merging.

**Pull request merged**